utils/utils.go

```diff
@@ -127,3 +130,22 @@ func ExtractVersionAndServiceFromUserAgent(userAgent string) (reqVer, service st
 	service = (strings.Split(userAgentSplit[0], "-"))[1]
 	return reqVer, service
 }
+
+func UrlWithFederation(remoteUrl string) (string, error) {
+	if param.Federation_DiscoveryUrl.IsSet() {
+		parsedUrl, err := url.Parse(remoteUrl)
+		if err != nil {
+			newErr := errors.New(fmt.Sprintf("error parsing source url: %s", err))
+			return "", newErr
+		}
+		if parsedUrl.Host != "" {
+			newErr := errors.New("Source URL should not have a host when the Federation_DiscoveryUrl is set")
```

Comment:
If we've configured a `Federation.DirectorUrl` in `pelican.yaml` but we run `pelican object get pelican://<some different federation>`, I think this will cause a failure. We should treat the provided Pelican URL's federation as an override.

Reply:
Ah, that's a good point.